### PR TITLE
[CAD-11] Add tagging support and tag search view

### PR DESCRIPTION
**Pull request by ruthwickp**
### Tag view on posts
Screen Shot 2021-03-18 at 11 36 20 AM

### Tag search view similar visual to post view
Screen Shot 2021-03-18 at 11 36 38 AM

### CODE REVIEW

**Comment by ruthwickp** [comment]
Put the navigation and tagging functionality behind a feature flag

File: config/routes.rb [hunk 1]
```diff
@@ -51,8 +51,14 @@
 
     resources :companies, only: [:edit, :update]
 
+    resources :tags, only: [:index]
+    post 'tags/search'
```
Comment (pranav7) [inline config/routes.rb]: You can probably write this as below.

```suggestion
    resources :tags, only: [:index] do
      get :search, on: :collection
    end
```

Also, should search be a `get` endpoint?

File: app/javascript/api/Tags.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import API from 'Services/api';
+
+class Tags {
+
+
```
Comment (pranav7) [inline app/javascript/api/Tags.js:5]:
```suggestion
```

File: app/javascript/components/admin/TagPostListItem.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      </div>
+      <div className="post-meta c-soft float-right">
+        {tags}
+      </div>
+    </a>
+  );
+};
+
+export default TagPostListItem;
+
```
Comment (pranav7) [inline app/javascript/components/admin/TagPostListItem.js]:
```suggestion
```

**Review by pranav7 — COMMENTED** [review]
Nice one, just a few nitpicks looks good otherwise

File: config/routes.rb (hunk 1 above)
Comment (ruthwickp) [reply config/routes.rb]: It's a `post` endpoint since I'm passing a JSON body to it.

Comment (pranav7) [reply config/routes.rb]: Ah cool.

**Review by pranav7 — APPROVED** [review]
Ship it 🚢

**Comment by pranav7** [comment]
Actually before you merge, we should have a quick chat.

**Pull request closed without merging**